modules/markets-api/src/main/scala/fi/spectrumlabs/markets/api/services/AnalyticsService.scala

```diff
@@ -1,40 +1,65 @@
 package fi.spectrumlabs.markets.api.services
 
 import cats.data.OptionT
-import cats.{Functor, Monad}
+import cats.{Functor, Monad, Parallel}
 import derevo.derive
-import fi.spectrumlabs.core.models.domain.{Amount, Pool}
+import fi.spectrumlabs.core.models.domain.{Amount, AssetClass, Pool, PoolFee, PoolId}
 import fi.spectrumlabs.markets.api.configs.MarketsApiConfig
-import fi.spectrumlabs.markets.api.models.PoolInfo
+import fi.spectrumlabs.markets.api.models.{PoolInfo, PoolOverview}
 import fi.spectrumlabs.markets.api.repositories.repos.{PoolsRepo, RatesRepo}
 import tofu.higherKind.Mid
 import tofu.higherKind.derived.representableK
 import tofu.logging.{Logging, Logs}
 import tofu.syntax.logging._
 import tofu.syntax.monadic._
+import cats.syntax.parallel._
+import cats.syntax.option._
 
 import scala.concurrent.duration.FiniteDuration
 import scala.math.BigDecimal.RoundingMode
 
 @derive(representableK)
 trait AnalyticsService[F[_]] {
+  def getPoolsOverview(period: FiniteDuration): F[List[PoolOverview]]
+
   def getPoolInfo(poolId: String, period: FiniteDuration): F[Option[PoolInfo]]
 }
 
 object AnalyticsService {
 
-  def create[I[_]: Functor, F[_]: Monad](config: MarketsApiConfig)(implicit
+  def create[I[_]: Functor, F[_]: Monad: Parallel](config: MarketsApiConfig)(implicit
     ratesRepo: RatesRepo[F],
     poolsRepo: PoolsRepo[F],
     logs: Logs[I, F]
   ): I[AnalyticsService[F]] =
     logs.forService[AnalyticsService[F]].map(implicit __ => new Tracing[F] attach new Impl[F](config))
 
-  final private class Impl[F[_]: Monad](config: MarketsApiConfig)(implicit
+  final private class Impl[F[_]: Monad: Parallel](config: MarketsApiConfig)(implicit
     ratesRepo: RatesRepo[F],
     poolsRepo: PoolsRepo[F]
   ) extends AnalyticsService[F] {
 
+    def getPoolsOverview(period: FiniteDuration): F[List[PoolOverview]] =
+      poolsRepo.getPools.flatMap(
+        _.parTraverse { p =>
+          getPoolInfo(p.poolId, period).map { info =>
+            for {
+              x <- AssetClass.fromString(p.x)
+              y <- AssetClass.fromString(p.y)
```

Comment:
Untyped data should not leak into service code. Invalid strings (p.x, p.y) should be rejected at the stage of parsing the inputs (`Get` instances in this case).